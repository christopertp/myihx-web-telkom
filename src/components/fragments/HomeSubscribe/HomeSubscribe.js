import React from 'react';
import { useSelector } from 'react-redux';
import Button from '../../elements/Button';
import useSlider from '../../../hooks/useSlider';
import { remToPx } from '../../../utils/unit';
import styles from './styles.scoped.css';

export default function HomeSubscribe() {
  const { isLoading, subscriptions } = useSelector(s => s.home);
  const slider = useSlider(3000, remToPx(1));
  return (
    <section className={styles.root}>
      {(
        isLoading.h || !subscriptions || <>
          <h4>Subscribe &amp; Watch</h4>
          <div className={styles.logos}>
            {subscriptions.vodImages.map((i, idx) => <img alt="vod" key={idx} src={i} />)}
          </div>
          <h2>{subscriptions.title}</h2>
          <p>{subscriptions.description}</p>
          <Button>Subscribe</Button>
          <ul {...slider}>
            {subscriptions.images.map((i, idx) => <img alt="" key={idx} src={i} />)}
          </ul>
        </>
      )}
    </section>
  );
}
