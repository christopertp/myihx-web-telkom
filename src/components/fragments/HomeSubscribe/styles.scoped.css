.root {
  background-color: #ffffff;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: repeat(5, auto);
  margin: 0 var(--neg-side-margin);
  padding: 4rem var(--side-margin) 2.5rem;
}

.root > h4 {
  color: #999999;
  font-size: 0.875rem;
  font-weight: normal;
  grid-column: 1;
  letter-spacing: 0.0625rem;
  margin-bottom: 2.5rem;
  text-transform: uppercase;
}

.logos {
  align-items: center;
  display: flex;
  grid-column: 1;
  margin-bottom: 1rem;
}

.logos > img { width: 4.625rem; }
.logos > img:nth-child(2) { margin: 0 1rem; }

.root > h2 {
  font-size: 1.75rem;
  grid-column: 1;
  margin-bottom: 0.75rem;
}

.root > p {
  color: #808080;
  grid-column: 1;
  line-height: 1.6;
  margin-bottom: 2rem;
}

.root > button {
  grid-column: 1;
  width: 13rem;
}

.root > ul {
  grid-column: 2;
  grid-row: 1/6;
  overflow-x: auto;
  margin-left: 1.25rem;
  scroll-snap-type: x mandatory;
  white-space: nowrap;
}

.root > ul > img {
  background-color: #999999;
  border-radius: 0.5rem;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  height: 23.5rem;
  scroll-snap-align: start;
}

.root > ul > img:not(:last-child) { margin-right: 1rem; }

@media only screen and (max-width: 48em) {
  .root {
    background-color: #f2f2f2;
    margin: 0 -1rem;
    grid-template-columns: auto;
    grid-template-rows: auto auto auto;
    padding: 1rem 1rem 1.25rem;
  }
  .root > h4 { margin-bottom: 1rem }
  .root > h2, .root > p, .logos { display: none; }
  .root > ul {
    grid-column: 1;
    grid-row: 2;
    margin-bottom: 1rem;
    margin-left: 0;
  }
  .root > ul > img {
    border-radius: 0.25rem;
    height: 8.25rem;
    margin-right: 0.75rem
  }
  .root > button {
    border-radius: 0.25rem !important;
    font-size: 0.75rem;
    grid-column: 1;
    grid-row: 3;
    width: 100%;
  }
}
