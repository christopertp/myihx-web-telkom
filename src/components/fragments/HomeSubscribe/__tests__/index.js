import React from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import HomeSubscribe from '../HomeSubscribe';

jest.mock('../../../../hooks/useSlider');

jest.mock('../../../../utils/unit', () => ({
  remToPx: v => v,
}));

useSelector.mockImplementation(fn => {
  fn({});
  return { isLoading: { h: false }, subscriptions: { images: [{}], vodImages: [{}], } };
});

describe('src/components/fragments/HomeSubscribe', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<HomeSubscribe />);
    expect(tree).toMatchSnapshot();
  });

  test('render loading', () => {
    useSelector.mockImplementationOnce(() => ({ isLoading: { h: true } }));
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<HomeSubscribe />);
    expect(tree).toMatchSnapshot();
  });
});
