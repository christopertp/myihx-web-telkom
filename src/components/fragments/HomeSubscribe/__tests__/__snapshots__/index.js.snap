// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fragments/HomeSubscribe render 1`] = `
<section>
  <React.Fragment>
    <h4>
      Subscribe & Watch
    </h4>
    <div>
      <img
        alt="vod"
        src={Object {}}
      />
    </div>
    <h2 />
    <p />
    <Button
      className=""
      disabled={false}
      onClick={null}
      to=""
      type="button"
      variant="red"
    >
      Subscribe
    </Button>
    <ul>
      <img
        alt=""
        src={Object {}}
      />
    </ul>
  </React.Fragment>
</section>
`;

exports[`src/components/fragments/HomeSubscribe render loading 1`] = `
<section>
  true
</section>
`;
